deployment/paiLibrary/paiOrchestration/temp_config.py

```diff
@@ -0,0 +1,68 @@
+# Copyright (c) Microsoft Corporation
+# All rights reserved.
+#
+# MIT License
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+# documentation files (the "Software"), to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+# to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+# BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+# DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+import os
+import time
+import yaml
+import logging
+
+from ..common import linux_shell
+from ...confStorage.download import download_configuration
+
+
+class temp_config:
+
+    def __init__(self, kube_config_path=None):
+        self._logger = logging.getLogger(__name__)
+        self._kube_config_path = None
+        if kube_config_path != None:
+            self._kube_config_path = os.path.expanduser(kube_config_path)
+        self._path = os.path.join(os.environ['HOME'], '.pai-config-' + str(time.time()))
+        self._pull_config_files()
+        self._generate_config_files()
+    
+    def _pull_config_files(self):
+        self._logger.info("Pull config from k8s cluster: `layout.yaml` and `services-configuration.yaml`")
+        get_handler = download_configuration(
+            config_output_path=self._path,
+            kube_config_path=self._kube_config_path
+        )
+        get_handler.run()
+    
+    def _generate_config_files(self):
+        self._logger.info("Generate config files: `hosts.yml` and `openpai.yml`")
+        linux_shell.execute_shell_raise(
+            shell_cmd="cd ./contrib/kubespray/ && python3 ./script/k8s_generator.py -l {} -c {} -o {}".format(
+                os.path.join(self._path, "layout.yaml"),
+                os.path.join(self._path, "services-configuration.yaml"),
```

Comment:
`-c` should be `config.yaml`. 
However, the problem is that `config.yaml` is not saved in k8s, please fix this in another PR.

Reply:
Change to `-s` here?

Reply:
> Change to `-s` here?

The problem is : Currently, when we call `paictl.py config push -p <> -m service` to update cluster configuration, only `layout.yaml` & `services-configuration.yaml` will be saved.  `config.yaml` should also be saved so that the admin users don't need to backup this file locally. 

I'm suggesting to fix this problem in another PR.